app/controllers/api/v1/packages_controller.rb

```diff
@@ -212,7 +212,13 @@ def package_valuation
       end
 
       def print_inventory_label
-        PrintLabelJob.perform_later(@package.id, User.current_user.id, "inventory_label", print_count)
+        printer = PrintersUser.where(user_id: User.current_user.id, tag: params[:tag]).first.try(:printer)
+        return render json: { errors: I18n.t("package.printer_not_found") }, status: 400 unless printer
+        opts = {
+          print_count: print_count,
+          label: "inventory_label"
+        }
+        PrintLabelJob.perform_later(@package.id, printer.id, opts)
```

Comment:
There is already an existing `print_inventory_label` method in the `packages_controller.rb`, probably you can enhance that to take in arguments

or even better move it to a model method
```
# from controller
printer.print_inventory_label(opts)

# in model 
def print_inventory_label
 PrintLabelJob.perform_later(slef.id, opts[:user_id], "inventory_label", opts[:print_count])
end
```